src/controllers/hooks/router-store/index.tsx

```diff
@@ -18,3 +31,43 @@ export const useRouterActions = (): RouterActionsType => {
 
   return allActions;
 };
+
+const useParam = createHook<
```

Comment:
This should be extended to be used from `usePathParam` too. We could make the selector accepting `{ paramKey, source }` or something similar so we can specify if we want the path parameter or the query parameter and avoid having another hook.

Reply:
I created two new two creators instead — `createQueryParamHook` and `createPathParamHook`